sympy/utilities/codegen.py

```diff
@@ -552,21 +556,56 @@ def routine(self, name, expr, argument_sequence, global_vars):
 
         """
 
+        if self.cse:
+            from sympy.simplify.cse_main import cse
+
+            if is_sequence(expr) and not isinstance(expr, (MatrixBase, MatrixExpr)):
+                if not expr:
+                    raise ValueError("No expression given")
+                assert all([e.is_Equality for e in expr]), "Lists of expressions must all be Equalities"
```

Comment:
change use of `assert` to an explicit raise of some exception (and those should be tested as well). We try to contain the use of `assert` to test files only (since python can be instructed to ignore assert statements).